bitcash/network/services.py

```diff
@@ -11,209 +12,56 @@
-        r.raise_for_status()  # pragma: no cover
-        return r.json()
-
-    @classmethod
-    def get_transactions(cls, address):
-        r = requests.get(cls.MAIN_ADDRESS_API.format(
-            address), timeout=DEFAULT_TIMEOUT)
-        r.raise_for_status()  # pragma: no cover
-        return r.json()['transactions']
-
-    @classmethod
-    def get_tx_amount(cls, txid, txindex):
-        r = requests.get(cls.MAIN_TX_AMOUNT_API.format(
-            txid), timeout=DEFAULT_TIMEOUT)
-        r.raise_for_status()  # pragma: no cover
-        response = r.json(parse_float=Decimal)
-        return (Decimal(response['vout'][txindex]['value']) * BCH_TO_SAT_MULTIPLIER).normalize()
-
-    @classmethod
-    def get_unspent(cls, address):
-        r = requests.get(cls.MAIN_UNSPENT_API.format(
-            address), timeout=DEFAULT_TIMEOUT)
-        r.raise_for_status()  # pragma: no cover
-        return [
-            Unspent(currency_to_satoshi(tx['amount'], 'bch'),
-                    tx['confirmations'],
-                    tx['scriptPubKey'],
-                    tx['txid'],
-                    tx['vout'])
-            for tx in r.json()
-        ]
-
-    @classmethod
-    def broadcast_tx(cls, tx_hex):  # pragma: no cover
-        r = requests.post(cls.MAIN_TX_PUSH_API, json={
-                          cls.TX_PUSH_PARAM: tx_hex, 'network': 'mainnet', 'coin': 'BCH'}, timeout=DEFAULT_TIMEOUT)
-        return True if r.status_code == 200 else False
-
-
-class BitcoinDotComAPI():
+class BitcoinDotComAPI:
     """ rest.bitcoin.com API """
-    MAIN_ENDPOINT = 'https://rest.bitcoin.com/v2/'
-    MAIN_ADDRESS_API = MAIN_ENDPOINT + 'address/details/{}'
-    MAIN_UNSPENT_API = MAIN_ENDPOINT + 'address/utxo/{}'
-    MAIN_TX_PUSH_API = MAIN_ENDPOINT + 'rawtransactions/sendRawTransaction/{}'
-    MAIN_TX_API = MAIN_ENDPOINT + 'transaction/details/{}'
-    MAIN_TX_AMOUNT_API = MAIN_TX_API
-    MAIN_RAW_API = MAIN_ENDPOINT + 'transaction/details/{}'
-    TX_PUSH_PARAM = 'rawtx'
 
-    TEST_ENDPOINT = 'https://trest.bitcoin.com/v2/'
-    TEST_ADDRESS_API = TEST_ENDPOINT + 'address/details/{}'
-    TEST_UNSPENT_API = TEST_ENDPOINT + 'address/utxo/{}'
-    TEST_TX_PUSH_API = TEST_ENDPOINT + 'rawtransactions/sendRawTransaction/{}'
-    TEST_TX_API = TEST_ENDPOINT + 'transaction/details/{}'
-    TEST_TX_AMOUNT_API = TEST_TX_API
-    TEST_RAW_API = TEST_ENDPOINT + 'transaction/details/{}'
-
-    REGTEST_ENDPOINT = 'http://localhost:12500/v2/'
-    REGTEST_ADDRESS_API = REGTEST_ENDPOINT + 'address/details/{}'
-    REGTEST_UNSPENT_API = REGTEST_ENDPOINT + 'address/utxo/{}'
-    REGTEST_TX_PUSH_API = REGTEST_ENDPOINT + 'rawtransactions/sendRawTransaction/{}'
-    REGTEST_TX_API = REGTEST_ENDPOINT + 'transaction/details/{}'
-    REGTEST_TX_AMOUNT_API = REGTEST_TX_API
-    REGTEST_RAW_API = REGTEST_ENDPOINT + 'transaction/details/{}'
+    NETWORK_ENDPOINTS = {
+        'mainnet': os.getenv("BITCOINCOM_API_MAINNET") or 'https://rest.bitcoin.com/v2/',
```

Comment:
I'm not sure about this concept in general, but you can do:

`os.getenv("env", "default")`

Reply:
I didn't realize `getenv` had a default. Awesome! Changed it: c02549c